### DISCUSSION TITLE
Keithley 2450 set up trouble-shooting

### DISCUSSION THREAD

**Discussion by Thayleez** [microsoft/Qcodes#6065] [Q&A] [answered]
I am realtively new to python and qcodes and am attempting to set up transistor measurements, using two Keithley 2450's and qcodes. 
My first question is: Is there anyone with experience with this or similar set up who can answer some most likely stupid questions ? 

My second question is: I have followed the qcodes 2450 example experiment and got it to work. If I simply then run my script again, I get a KeyError: 'Another instrument has the name: gate'. My assumption is that I need a line of code which somehow resets the instrument name at the beginning of the experiment but I can't seem to find this. Thanks for any help!

**Comment by jenshnielsen** [comment]
I don't have any specific experience with the 2450 but when you connect to a qcodes instrument it will stay connected until you call close on the instrument class or close that python session. That is by design such that you don't have to spend time connecting and getting the state of the instrument more than once. 

Usually people setup their scripts such that they connect once and then reuse that instrument. But you may also chose to call close explicitly at the end of your script

**Reply by jenshnielsen** [reply]
For reference I did toy around with making the constructor reuse an existing instrument in https://github.com/microsoft/Qcodes/pull/4038/ However, its not something that I really have cycles to look into at the moment.

**Comment by EdwardLaird1** [comment, marked as answer]
I use this instrument and my setup script includes qc.Instrument.close_all() just before I begin setting up my station, to do exactly what Jens suggests.

**Reply by Thayleez** [reply]
Very much appreciated, thank you.
